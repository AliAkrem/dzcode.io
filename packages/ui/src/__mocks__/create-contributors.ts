import { ContributorEntity } from "@dzcode.io/models/dist/contributor";
import { v4 as uuid } from "uuid";

export const createContributors = (number: number) => {
  const contributors: ContributorEntity[] = [];
  for (let i = 0; i < number; i++) {
    contributors.push({
      id: uuid(),
      avatarUrl: `mock-avatar-${i}`,
      username: `mock-user-${i}`,
      repositories: [
        {
          owner: `user-${i}`,
          provider: `github`,
          repository: `repository-${i}`,
        },
      ],
    });
  }
  return contributors;
};
