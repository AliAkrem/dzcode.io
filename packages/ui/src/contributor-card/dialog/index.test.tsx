import { render, screen } from "@testing-library/react";
import { createRepositories } from "src/__mocks__/create-repositories";

import { ContributionsDialog, ContributionsDialogProps } from ".";

it("should render a closed dialog on first mount", async () => {
  const onClose = jest.fn();
  const props: ContributionsDialogProps = {
    onClose: onClose,
    open: false,
    repositories: createRepositories(2),
    repositoriesText: "Repositories",
  };

  const { container } = render(<ContributionsDialog {...props} />);
  expect(container).toMatchSnapshot();
});

it("should handle users with no projects", async () => {
  const onClose = jest.fn();
  const props: ContributionsDialogProps = {
    onClose: onClose,
    open: true,
    repositories: createRepositories(2),
    repositoriesText: "Repositories",
  };

  render(<ContributionsDialog {...props} />);
  const dialog = await screen.findByTestId("contribution-dialog");
  expect(dialog).toBeInTheDocument();
});

it("should handle users with no repositories", async () => {
  const onClose = jest.fn();
  const props: ContributionsDialogProps = {
    onClose: onClose,
    open: true,
    repositories: [],
    repositoriesText: "Repositories",
  };

  render(<ContributionsDialog {...props} />);
  const dialog = await screen.findByTestId("contribution-dialog");
  expect(dialog).toBeInTheDocument();
});

it("should render a dialog", async () => {
  const onClose = jest.fn();
  const props: ContributionsDialogProps = {
    onClose: onClose,
    open: true,
    repositories: createRepositories(2),
    repositoriesText: "Repositories",
  };

  render(<ContributionsDialog {...props} />);
  const dialog = await screen.findByTestId("contribution-dialog");
  expect(dialog).toBeVisible();
  expect(dialog).toBeInTheDocument();
});
