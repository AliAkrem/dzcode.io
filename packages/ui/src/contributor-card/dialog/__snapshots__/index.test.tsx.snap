// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a closed dialog on first mount 1`] = `<div />`;
